src/pubsub/src/lib.rs

```diff
@@ -66,6 +66,47 @@ pub mod model_ext {
     pub use crate::publisher::model_ext::*;
 }
 
+/// Clients to interact with Google Cloud Pub/Sub.
+///
+/// This module contains the primary entry points for the library, including
+/// clients for publishing messages and managing topics and subscriptions.
+///
+/// # Example: Publishing Messages
+///
+/// ```
+/// # async fn sample() -> anyhow::Result<()> {
+/// use google_cloud_pubsub::client::PublisherFactory;
+/// use google_cloud_pubsub::model::PubsubMessage;
+///
+/// // Create a factory for creating publishers.
+/// let factory = PublisherFactory::builder().build().await?;
+///
+/// // Create a publisher that handles batching for a specific topic.
+/// let publisher = factory.publisher("projects/my-project/topics/my-topic").build();
```

Comment:
For a future PR: I think we want a variant where we do:

```suggestion
/// let publisher: BatchedPublisher =
///     Publisher::builder("projects/my-project/topics/my-topic").build().await?;
```

And separately:

```rust
/// let publisher: Publisher = Publisher::shared().build().await?;
/// let batched: BatchedPublisher =
///     publisher.builder("projects/my-project/topics/my-topic").build();
```

and someday:

```rust
/// let publisher: Publisher = Publisher::shared().build().await?;
/// publisher.unbatched_publish("projects/my-project/topics/my-topic", message).send().await?;
```